### DISCUSSION TITLE
Queries related to SDK design rationales ?

### DISCUSSION THREAD

**Discussion by renjithrajagopal-sudo** [COVESA/covesa-aosp-sdk#2] [General]
Hey guys,
Thank you for great effort in pushing the SDK out of door under COVESA community.Have got couple of questions most of them being subjective.

Understanding is that service created here is to be allowing 3Ps to access(bi-directional) vehicle data without using Car API due to current Android framework access control model (protection level being system or platform) & additional device allowlisting(https://source.android.com/docs/core/permissions/perms-allowlist) done by OEM, If yes Can you address following ? 

1. How do OEM restrict API usage towards certain 3Ps ? Any access control model planned like RBAC ?
2. Will this work with GAS & Google play listing ?
3. Will the updates to library be available via Device OTA or updatable via playstore?
4. Any possibility of having it available as Vendor APEX in future ?
5. Backend data provider seems VHAL, So this makes service to run as system service & requires additional binder  MAC policies(SELinux) ?
6. Could this be implemented as Java SDK library so visibility could be controlled (public, system, testing)  ? https://source.android.com/docs/setup/build/java-library

Thanks

**Comment by JuhaniLehtimaeki** [comment]
> Will this work with GAS & Google play listing ?

Yes. While in the future some of the features might be replacement to what GAS provides, the library should stay compatible with GAS and non-GAS AAOS systems.
